### PR TITLE
Updating caching handlers

### PR DESCRIPTION
**Pull request by lakhveerk**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

Updated `app.lifecycle.registerBeforeSuspendOrTerminateHandler` to be asynchronous,
Updated `app.lifecycle.registerOnResumeHandler` to accept a handler type that changes `contentUrl` from string to URL object.

> If this Pull Request should close/resolve any issues when merged, use [the special syntax for that](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) here.

### Main changes in the PR:

1. Updated `app.lifecycle.registerBeforeSuspendOrTerminateHandler` to be asynchronous
2. Updated `app.lifecycle.registerOnResumeHandler` to accept a handler type that changes `contentUrl` from string to URL object

### Validation performed:

1. <Step 1>
2. <Step 2>

### Unit Tests added:

> Unit tests are required for all changes. If no unit tests were added as part of this change, please explain why they aren't necessary.

<Yes/No>

### End-to-end tests added:

<Yes/No>
They are coming in host sdk PR

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

<Yes/No>
yes

### CODE REVIEW

File: change/@microsoft-teams-js-ec870a0b-890d-4ada-b33b-301c1c28a8d3.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Updated `app.lifecycle.registerBeforeSuspendOrTerminateHandler` to be asynchronous, and updated `app.lifecycle.registerOnResumeHandler` to accept a new Handler type, changing `contentUrl` from string to URL object.",
```
Comment (AE-MS) [inline change/@microsoft-teams-js-ec870a0b-890d-4ada-b33b-301c1c28a8d3.json:3]: >from string to URL object [](http://example.com/codeflow?start=202&length=25)

from `string` to `URL` object

**Review by AE-MS — APPROVED** [review]

**Review by TrevorJoelHarris — APPROVED** [review]

**Comment by migueljolv** [comment]
Hello @TrevorJoelHarris @AE-MS,
I'm curious, does this PR enable the App Caching mechanism in the new MS Teams application?

**Comment by AE-MS** [comment]
Unfortunately, it does not; it only updates the platform pieces. I believe issue #2121 is tracking the work that the Teams team has to do to enable that. I'm guessing you are already aware of that issue. 😉

**Comment by migueljolv** [comment]
> Unfortunately, it does not; it only updates the platform pieces. I believe issue #2121 is tracking the work that the Teams team has to do to enable that. I'm guessing you are already aware of that issue. 😉

I see, thanks for the quick reply 😄

**Pull request merged**